crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4548,6 +4548,31 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn simplify_true_and_false_to_bool() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_dedented(
+            "src/a.py",
+            "
+            from typing_extensions import reveal_type
+
+            def returns_bool() -> bool: ...
```

Comment:
Ah, at some point we'll start checking return types, which might lead to us emitting a spurious diagnostic in this test snippet. We can make it robust to future improvements like so:

```suggestion
            def returns_bool() -> bool:
                return True
```